app/lib/screens/identity_verification_screen.dart

```diff
@@ -1035,29 +1076,50 @@ class _IdentityVerificationScreenState
                                   step == 2 &&
                                           Globals().hidePhoneButton.value ==
                                               true
-                                      ? Row(
-                                          children: <Widget>[
-                                            Text(
-                                              'SMS sent, retry in ${calculateMinutes()} minute${calculateMinutes() == '1' ? '' : 's'}',
-                                              overflow: TextOverflow.clip,
-                                              style: Theme.of(context)
-                                                  .textTheme
-                                                  .bodySmall!
-                                                  .copyWith(
-                                                      fontWeight:
-                                                          FontWeight.bold,
-                                                      color: Theme.of(context)
-                                                          .colorScheme
-                                                          .warning),
-                                            )
-                                          ],
-                                        )
+                                      ? Row(children: <Widget>[
+                                          ValueListenableBuilder<int>(
+                                            valueListenable:
+                                                phoneCountdownNotifier,
+                                            builder: (context, remainingTime,
+                                                child) {
+                                              if (remainingTime > 0) {
+                                                String formattedTime;
+                                                if (remainingTime >= 60) {
+                                                  int minutes = remainingTime ~/
+                                                      60;
+                                                  int seconds = remainingTime %
+                                                      60;
+                                                  formattedTime =
+                                                      '${minutes}m ${seconds}s';
+                                                } else {
+                                                  formattedTime =
+                                                      '${remainingTime}s';
+                                                }
```

Comment:
Can we export the logic here in an outer function?

Reply:
done